yt_dlp/extractor/canalsurmas.py

```diff
@@ -0,0 +1,143 @@
+        - badges
+        - type
+        - duration
+        - clasification
+        - is_premium
+        - inapppurchase
+        - background
+        - description
+        - short_description
+        - genre
+        - emision_date
+        - quality
+        - has_subtitle
+        - technical_details
+        - trailer
+        - title_image
+        - is_ads
+        - has_dvr
+        - next_chapter
+        - reference
+          - A string value (e.g. "0000051497") that is needed when sending a request to get the main M3U8 file of the video.
+        - season_category
+        - main_category
+          - A nested object that contains the following fields:
+            - id
+            - name
+            - image
+            - badges
+            - type
+            - duration
+            - clasification
+            - is_premium
+            - inapppurchase
+            - background
+            - description
+            - short_description
+            - genre
+            - emision_date
+            - quality
+            - has_subtitle
+            - technical_details
+            - trailer
+            - title_image
+            - reference
+            - start_second_chapter
+            - finish_second_chapter
+            - alternative_image
+            - image_mobile
+            - image_medium
+            - image_tiny
+            - background_medium
+            - background_tiny
+            - alternative_image_medium
+            - alternative_image_tiny
+            - image_mobile_medium
+            - image_mobile_tiny
+        - start_second_chapter
+        - finish_second_chapter
+        - created_at
+        - image_medium
+        - alternative_image
+        - alternative_image_medium
+        - image_mobile
+        - expiration_date
+        - publish_date
+        - is_freemium
+        - tags
+        '''
+        video_info = self._download_json(
+            url_or_request=f'https://api-rtva.interactvty.com/api/2.0/contents/content/{video_id}/?optional_fields=reference,description,main_category',
+            video_id=video_id,
+            headers={
+                'Authorization': f'jwtok {access_token}',
+            },
+        )
+
+        video_reference_num: str = video_info['reference']
+
+        video_main_category_name_as_lower_case_underscore_delimited_string = video_info['main_category']['name'].lower().replace(' ', '_')
+
+        # Another request can be made like so to get the main M3U8 file for a video with a specific reference number.
+        # The returned M3U8 file points to other M3U8 files for different qualities (i.e. 480p, 1080p) and resolutions.
+        # Example URL: https://cdn.rtva.interactvty.com/archivo_ott/el_gran_queo/0000051497/0000051497_3R.m3u8
+        video_master_playlist_m3u8_url = f'https://cdn.rtva.interactvty.com/archivo_ott/{video_main_category_name_as_lower_case_underscore_delimited_string}/{video_reference_num}/{video_reference_num}_3R.m3u8'
```

Comment:
Instead of constructing the URL like this I would call the `content_resources` api
```
https://api-rtva.interactvty.com/api/2.0/contents/content_resources/222388/?optional_fields=type,videotype,media_url,is_initial,assets,drm_token,test,plain,url_app_web,thumbnail_track_url,product,remote_product_id
```